app/cmd/plugin_upgrade.go

```diff
@@ -104,3 +116,25 @@ func (p *PluginUpgradeOption) findUpgradeablePlugins(jclient *client.PluginManag
 	}
 	return
 }
+
+func (p *PluginUpgradeOption) findCompatiblePlugins(installedPlugins []client.InstalledPlugin) (plugins []string) {
```

Comment:
We need to keep to use the same rules with Jenkins UI operation. Could you please give us a reference or link which is about the rules? In addition, if there're 50 plugins, then this command needs to create 51 HTTP connections. It's too bad. Always try not to do it connect in a loop.